tests/bluetooth/common/testlib/include/bs_helpers/flags.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/atomic.h>
+
+#define DECLARE_FLAG(flag) extern atomic_t flag
+#define DEFINE_FLAG(flag)  atomic_t flag = (atomic_t) false
+#define SET_FLAG(flag)     (void)atomic_set(&flag, (atomic_t) true)
+#define UNSET_FLAG(flag)   (void)atomic_set(&flag, (atomic_t) false)
+
+#define IS_FLAG_SET(flag) (atomic_get(&flag) != false)
+
+#define WAIT_FOR_VAL(var, val)                                                                     \
```

Comment:
Please mention all these macros can only be called from Zephyr threads (and not from a bs_tests ticker callback)